dirty_cat/test/test_deduplicate.py

```diff
@@ -0,0 +1,119 @@
+import string
+from typing import List
+
+import numpy as np
+import pytest
+from scipy.cluster.hierarchy import linkage
+from scipy.spatial.distance import squareform
+
+from dirty_cat.deduplicate import (
+    compute_ngram_distance,
+    create_spelling_correction,
+    deduplicate,
+    guess_clusters,
+)
+
+
+def generate_example_data(examples, entries_per_example, prob_mistake_per_letter):
+    """Helper function to generate data consisting of multiple entries per example.
+    Characters are misspelled with probability `prob_mistake_per_letter`"""
+
+    data = []
+    for example, n_ex in zip(examples, entries_per_example):
+        len_ex = len(example)
+        # generate a 2D array of chars of size (n_ex, len_ex)
+        str_as_list = np.array([list(example)] * n_ex)
+        # randomly choose which characters are misspelled
+        idxes = np.where(
+            np.random.random(len(example[0]) * n_ex) < prob_mistake_per_letter
```

Comment:
You need to use a seeded local random number generator: at the beginning of the function, define
```python
rng = np.random.RandomState(seed)
```
and use it's methods